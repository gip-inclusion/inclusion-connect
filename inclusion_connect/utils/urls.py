from urllib.parse import parse_qsl, urlparse

from django.utils.http import url_has_allowed_host_and_scheme, urlencode


def add_url_params(url: str, params: dict[str, str]) -> str:
    """Add GET params to provided URL being aware of existing.

    :param url: string of target URL
    :param params: dict containing requested params to be added
    :return: string with updated URL

    >> url = 'http://localhost:8000/login/activate_siae_staff_account?next_url=%2Finvitations
    >> new_params = {'test': 'value' }
    >> add_url_params(url, new_params)
    'http://localhost:8000/login/activate_siae_staff_account?next_url=%2Finvitations&test=value
    """

    # Remove params with None values
    params = {key: params[key] for key in params if params[key] is not None}
    url_parts = urlparse(url)
    query = get_url_params(url)
    query.update(params)

    new_url = url_parts._replace(query=urlencode(query)).geturl()

    return new_url


def get_url_params(url: str) -> dict[str, str]:
    return dict(parse_qsl(urlparse(url).query))


def is_inclusion_connect_url(request, url):
    return url_has_allowed_host_and_scheme(url, request.get_host(), require_https=request.is_secure())
