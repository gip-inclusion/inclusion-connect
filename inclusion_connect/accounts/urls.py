from django.urls import path, re_path

from inclusion_connect.accounts import views


app_name = "accounts"

urlpatterns = [
    re_path(r"^login/$", views.LoginView.as_view(), name="login"),
    re_path(r"^register/$", views.RegisterView.as_view(), name="register"),
    re_path(r"^activate/$", views.ActivateAccountView.as_view(), name="activate"),
    re_path(r"^password_reset/$", views.PasswordResetView.as_view(), name="password_reset"),
    path("reset/<uidb64>/<token>/", views.PasswordResetConfirmView.as_view(), name="password_reset_confirm"),
    re_path(r"^accept-terms/$", views.AcceptTermsView.as_view(), name="accept_terms"),
    re_path(r"^my-account/$", views.EditUserInfoView.as_view(), name="edit_user_info"),
    re_path(r"^change-password/$", views.PasswordChangeView.as_view(), name="change_password"),
]
