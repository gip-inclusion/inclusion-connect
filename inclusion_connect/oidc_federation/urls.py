from django.urls import include, re_path

from . import peama


app_name = "oidc_federation"

peama_urlpatterns = [
    re_path(r"^callback/", peama.OIDCAuthenticationCallbackView.as_view(), name="callback"),
    re_path(r"^authenticate/", peama.OIDCAuthenticationRequestView.as_view(), name="init"),
    # There's no logout available on PEAMA
]

urlpatterns = [
    re_path(r"^peama/", include((peama_urlpatterns, "peama"))),
]
