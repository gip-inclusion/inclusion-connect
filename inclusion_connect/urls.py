"""inclusion_connect URL Configuration

The `urlpatterns` list routes URLs to views. For more information please see:
    https://docs.djangoproject.com/en/4.1/topics/http/urls/
Examples:
Function views
    1. Add an import:  from my_app import views
    2. Add a URL to urlpatterns:  path('', views.home, name='home')
Class-based views
    1. Add an import:  from other_app.views import Home
    2. Add a URL to urlpatterns:  path('', Home.as_view(), name='home')
Including another URLconf
    1. Import the include() function: from django.urls import include, path
    2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
"""
from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path


urlpatterns = [
    # Admin
    path("admin/", admin.site.urls),
    # Login and registration urls
    re_path(r"^accounts/", include("inclusion_connect.accounts.urls")),
    re_path(r"^accounts/", include("django.contrib.auth.urls")),
    # OIDC urls
    re_path(r"^auth/", include("inclusion_connect.oidc_overrides.urls")),
    re_path(r"^auth/", include("oauth2_provider.urls", namespace="oauth2_provider")),
]

for realm in settings.KEYCLOAK_REALMS:
    urlpatterns.append(
        re_path(
            rf"^realms/{realm}/protocol/openid-connect/",
            include("inclusion_connect.keycloak_compat.urls", namespace=f"keycloak_compat_{realm}"),
        )
    )

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
