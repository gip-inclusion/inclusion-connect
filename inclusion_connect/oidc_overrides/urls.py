from django.urls import re_path
from oauth2_provider import views as oauth2_views

from inclusion_connect.oidc_overrides import views


app_name = "oauth2_provider"

urlpatterns = [
    re_path(r"^authorize/", views.AuthorizationView.as_view(), name="authorize"),
    re_path(r"^register/", views.RegistrationView.as_view(), name="register"),
    re_path(r"^activate/", views.ActivationView.as_view(), name="activate"),
    re_path(r"^token/$", oauth2_views.TokenView.as_view(), name="token"),
    re_path(r"^revoke_token/$", oauth2_views.RevokeTokenView.as_view(), name="revoke-token"),
    re_path(r"^introspect/$", oauth2_views.IntrospectTokenView.as_view(), name="introspect"),
    # OIDC urls
    re_path(
        r"^\.well-known/openid-configuration/$",
        oauth2_views.ConnectDiscoveryInfoView.as_view(),
        name="oidc-connect-discovery-info",
    ),
    re_path(r"^\.well-known/jwks.json$", oauth2_views.JwksInfoView.as_view(), name="jwks-info"),
    re_path(r"^userinfo/$", oauth2_views.UserInfoView.as_view(), name="user-info"),
    re_path(r"^logout/", views.LogoutView.as_view(), name="logout"),
]
