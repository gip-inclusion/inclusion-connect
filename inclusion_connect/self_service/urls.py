from django.urls import re_path

from inclusion_connect.self_service import views


app_name = "self_service"

urlpatterns = [
    # Application management views
    re_path(r"^applications/$", views.ApplicationList.as_view(), name="list"),
    re_path(r"^applications/register/$", views.ApplicationRegistration.as_view(), name="register"),
    re_path(r"^applications/(?P<pk>[\w-]+)/$", views.ApplicationDetail.as_view(), name="detail"),
    re_path(r"^applications/(?P<pk>[\w-]+)/delete/$", views.ApplicationDelete.as_view(), name="delete"),
    re_path(r"^applications/(?P<pk>[\w-]+)/update/$", views.ApplicationUpdate.as_view(), name="update"),
]
