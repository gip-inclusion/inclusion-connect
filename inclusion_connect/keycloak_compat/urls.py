from django.urls import re_path
from oauth2_provider import views as oauth2_views

from ..accounts.views import EditUserInfoView
from ..oidc_overrides.views import AuthorizationView, LogoutView, RegistrationView
from . import views


app_name = "keycloak_compat"


urlpatterns = [
    re_path(
        r"^\.well-known/openid-configuration/$",
        oauth2_views.ConnectDiscoveryInfoView.as_view(),
        name="oidc-connect-discovery-info",
    ),
    re_path(r"^protocol/openid-connect/userinfo$", oauth2_views.UserInfoView.as_view(), name="user-info"),
    re_path(r"^protocol/openid-connect/auth$", AuthorizationView.as_view(), name="authorize"),
    re_path(r"^protocol/openid-connect/registrations$", RegistrationView.as_view(), name="registrations"),
    re_path(r"^protocol/openid-connect/token$", oauth2_views.TokenView.as_view(), name="token"),
    re_path(r"^protocol/openid-connect/logout$", LogoutView.as_view(), name="logout"),
    re_path(r"^account$", EditUserInfoView.as_view(), name="edit_user_info"),
    re_path(r"^login-actions/action-token$", views.ActionToken.as_view(), name="action-token"),
]
