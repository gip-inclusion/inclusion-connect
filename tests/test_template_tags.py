from bs4 import BeautifulSoup
from django.urls import reverse


def test_new_password_display(client, snapshot):
    response = client.get(reverse("accounts:register"))
    soup = BeautifulSoup(response.content, "html5lib", from_encoding=response.charset)
    form_group = soup.find("label", attrs={"for": "id_password1"}).parent
    assert str(form_group) == snapshot(name="new password has instructions")
