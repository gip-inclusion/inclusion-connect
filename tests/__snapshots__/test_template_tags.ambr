# serializer version: 1
# name: test_new_password_display[new password has instructions]
  '''
  <div class="mb-3 form-group-required"><label class="form-label" for="id_password1">Mot de passe</label><div class="input-group"><input autocomplete="new-password" class="form-control password-with-instructions" id="id_password1" name="password1" placeholder="**********" required="" type="password"/>
          <span class="input-group-text p-0">
              <button class="btn btn-sm btn-link btn-ico" data-it-password="toggle" type="button">
                  <i class="ri-eye-line"></i>
                  <span>Afficher</span>
              </button>
          </span>
          </div></div>
  '''
# ---
